connectors/sources/sharepoint_online.py

```diff
@@ -261,6 +287,19 @@ async def scroll(self, url):
             else:
                 break
 
+    async def scroll_with_delta_link(self, url):
+        scroll_url = url
+
+        while True:
```

Comment:
Can probably be `while scroll_url` to indicate the condition of breaking the loop better